nucypher/crypto/api.py

```diff
@@ -199,6 +206,29 @@ def generate_self_signed_certificate(host: str,
     return cert, private_key
 
 
+def generate_teacher_certificate(checksum_address: str, *args, **kwargs):
+    cert = __generate_self_signed_certificate(checksum_address=checksum_address, *args, **kwargs)
+    return cert
+
+
+def generate_self_signed_certificate(*args, **kwargs):
+    if 'checksum_address' in kwargs:
+        raise ValueError("checksum address cannot be used to generate standard self-signed certificates.")
+    cert = __generate_self_signed_certificate(checksum_address=None, *args, **kwargs)
+    return cert
+
+
+def read_certificate_pseudonym(certificate: Certificate):
+    try:
+        pseudonym = certificate.subject.get_attributes_for_oid(NameOID.PSEUDONYM)[0]
+    except IndexError:
+        raise RuntimeError("Invalid teacher certificate encountered: No checksum address present as pseudonym.")
```

Comment:
I know that `RuntimeError` is in line with what was done previously,  any chance we could modify the exception to be `ValueError`? same for the invalid checksum check below?